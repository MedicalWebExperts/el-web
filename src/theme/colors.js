export default {
  default: '#000000',
  primary: '#0000ff',
  secondary: '#00b33c',
};
