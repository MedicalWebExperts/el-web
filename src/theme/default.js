export default {
  colors: {
    primary: '#0000ff',
    secondary: '#00b33c',
    default: '#000000',
  },
  font: {
    primary: 'Georgia, serif',
    secondary: 'sans-serif',
  },
};
