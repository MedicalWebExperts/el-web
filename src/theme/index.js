import { withTheme } from 'theming';
import ThemeProvider from './provider';
import DefaultTheme from './default';

module.exports = {
  ThemeProvider,
  withTheme,
  DefaultTheme,
};
