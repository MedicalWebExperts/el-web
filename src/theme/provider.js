import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider as ThemeProviderBase } from 'theming';
import DefaultTheme from './default';

const propTypes = {
  children: PropTypes.element.isRequired,
  theme: PropTypes.oneOfType([PropTypes.shape({})]),
};

const defaultProps = {
  theme: {},
};

const ThemeProvider = ({ children, theme }) => {
  // check a util funcion for deep clone/merge
  const mergedTheme = Object.keys(theme).length === 0 ? DefaultTheme : theme;

  return (
    <ThemeProviderBase theme={mergedTheme}>
      {children}
    </ThemeProviderBase>
  );
};

ThemeProvider.propTypes = propTypes;
ThemeProvider.defaultProps = defaultProps;

export default ThemeProvider;
