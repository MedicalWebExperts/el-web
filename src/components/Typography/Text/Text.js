import React from 'react';
import PropTypes from 'prop-types';
import { withTheme } from '../../../theme';

const propTypes = {
  children: PropTypes.node,
  text: PropTypes.string,
  color: PropTypes.oneOf(['primary', 'secondary', 'default']),
  gutterBottom: PropTypes.number,
  align: PropTypes.oneOf(['left', 'right', 'center', 'inherit', 'justify']),
  noWrap: PropTypes.bool,
  secondary: PropTypes.bool,
  theme: PropTypes.shape({}).isRequired,
};

const defaultProps = {
  children: '',
  text: '',
  color: 'default',
  align: 'inherit',
  gutterBottom: 0,
  noWrap: false,
  secondary: false,
};

const Text = ({
  children,
  text,
  color,
  gutterBottom,
  align,
  noWrap,
  secondary,
  theme,
}) => (
  <p>
    <style jsx>{`
      p {
        color: ${theme.colors[color]}
        text-align: ${align}
        font-family: ${secondary ? theme.font.secondary : theme.font.primary}
        white-space: ${noWrap ? 'nowrap' : 'normal'}
        line-height: ${gutterBottom}
      }
    `}
    </style>
    {text || children}
  </p>
);

Text.propTypes = propTypes;
Text.defaultProps = defaultProps;
export default withTheme(Text);
