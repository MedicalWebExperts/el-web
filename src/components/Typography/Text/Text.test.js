import React from 'react';
import renderer from 'react-test-renderer';

import Text from './Text';
import { ThemeProvider } from '../../../theme';

describe('Component Text Snapshot', () => {
  it('without arguments, take the default values, use childreen prop', () => {
    const tree = renderer
      .create(<ThemeProvider><Text>This is a component Text!</Text></ThemeProvider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('use align prop with value right', () => {
    const tree = renderer
      .create(<ThemeProvider><Text align="right">This is a component Text!</Text></ThemeProvider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('secondary font-family of the theme', () => {
    const tree = renderer
      .create(<ThemeProvider><Text secondary>This is a component Text!</Text></ThemeProvider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('primary color of the theme', () => {
    const tree = renderer
      .create(<ThemeProvider><Text color="primary">This is a component Text!</Text></ThemeProvider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('use gutterBottom prop', () => {
    const tree = renderer
      /* eslint-disable */
      .create(<ThemeProvider><Text gutterBottom={20}>This is a component Text!</Text></ThemeProvider>)
      /* eslint-enable */
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('use gutterBottom prop and secondary color', () => {
    const tree = renderer
      .create(<ThemeProvider><Text gutterBottom={20} color="secondary">This is a component Text!</Text></ThemeProvider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('use gutterBottom prop and secondary font', () => {
    const tree = renderer
      /* eslint-disable */
      .create(<ThemeProvider><Text gutterBottom={20} secondary>This is a component Text!</Text></ThemeProvider>)
      /* eslint-enable */
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('use gutterBottom prop, secondary font and align', () => {
    const tree = renderer
      .create(<ThemeProvider><Text gutterBottom={20} secondary align="right">This is a component Text!</Text></ThemeProvider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('added optional style inline prop', () => {
    const myStyles = {
      marginTop: 30,
      lineHeight: 2,
    };
    const tree = renderer
      /* eslint-disable */
      .create(<ThemeProvider><Text styles={myStyles}>This is a component Text!</Text></ThemeProvider>)
      /* eslint-enable */
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
