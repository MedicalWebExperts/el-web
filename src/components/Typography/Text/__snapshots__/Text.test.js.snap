// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Text Snapshot primary color of the theme 1`] = `
<p
  className="jsx-395920955"
>
  This is a component Text!
</p>
`;

exports[`Component Text Snapshot secondary font-family of the theme 1`] = `
<p
  className="jsx-440853637"
>
  This is a component Text!
</p>
`;

exports[`Component Text Snapshot use align prop with value right 1`] = `
<p
  className="jsx-32005758"
>
  This is a component Text!
</p>
`;

exports[`Component Text Snapshot use gutterBottom prop 1`] = `
<p
  className="jsx-3903326633"
>
  This is a component Text!
</p>
`;

exports[`Component Text Snapshot use gutterBottom prop and secondary color 1`] = `
<p
  className="jsx-660067240"
>
  This is a component Text!
</p>
`;

exports[`Component Text Snapshot use gutterBottom prop and secondary font 1`] = `
<p
  className="jsx-997711959"
>
  This is a component Text!
</p>
`;

exports[`Component Text Snapshot use gutterBottom prop, secondary font and align 1`] = `
<p
  className="jsx-3149066066"
>
  This is a component Text!
</p>
`;

exports[`Component Text Snapshot without arguments, take the default values, use childreen prop 1`] = `
<p
  className="jsx-3729570811"
>
  This is a component Text!
</p>
`;
